build-system/status-page/incident.js

```diff
@@ -0,0 +1,170 @@
+    Please allow up to 30 minutes for the CDN to pick up the fix.`,
+  'resolved': 'The fix has been verified.',
+};
+
+/**
+ * Create incident on status.amp.dev
+ * @param {Array<string>} channels
+ * @param {Array<string>} formats
+ * @param {string} status
+ * @return {Promise<Object>} response
+ */
+async function createIncident(channels, formats, status) {
+  const components = {};
+  formats.forEach((format) => {
+    components[componentsIds[format]] = 'degraded_performance';
+  });
+
+  const incident = {
+    'name': `Incident in ${channels.join(' and ').toUpperCase()}`,
+    'status': status,
+    'impact_override': 'minor',
+    'body': `We are investigating reports of a bug that is seen in ${channels
+      .join(' and ')
+      .toUpperCase()}.
+      https://github.com/ampproject/amphtml/issues/${number}`,
+    'components': components,
+    'metadata': {
+      'github': {
+        'cherry_pick_issue_number': number,
+      },
+    },
+  };
+
+  const response = await fetch(`${apiUrl}/incidents`, {
+    method: 'POST',
+    headers,
+    body: JSON.stringify({'incident': incident}),
+  });
+
+  return await response.json();
+}
+
+/**
+ * Get unresolved incident by cherry-pick issue number
+ * @return {Promise<Object>}
+ */
+async function getIncident() {
+  const response = await fetch(`${apiUrl}/incidents/unresolved`, {
+    headers,
+  });
+
+  for (const incident of await response.json()) {
+    if (incident.metadata.github.cherry_pick_issue_number === number) {
+      return incident;
+    }
+  }
+
+  throw new Error(
+    `Could not find an unresolved incident for cherry-pick issue #${number}`
+  );
+}
+
+/**
+ * Updates incident on status.amp.dev
+ * @param {Array<string>} formats
+ * @param {string} status
+ * @return {Promise<Object>}
+ */
+async function updateIncident(formats, status) {
+  const components = {};
+  formats.forEach((format) => {
+    components[componentsIds[format]] =
+      status === 'resolved' ? 'operational' : 'degraded_performance';
+  });
+
+  const incident = await getIncident();
+  incident.body = updateBodies[status];
+  incident.status = status;
+  incident.components = components;
+
+  const response = await fetch(`${apiUrl}/incidents/${incident.id}`, {
+    method: 'PATCH',
+    headers,
+    body: JSON.stringify({'incident': incident}),
+  });
+
+  return await response.json();
+}
+
+/**
+ * Syncs incident with cherry-pick progress
+ * @return {Promise<any>}
+ */
+async function syncIncident() {
+  const formats = getFormats(body);
+  const channels = getChannels(body);
+  if (channels.length == 0) {
+    return;
+  }
+
+  // get step that was checked off
+  let checked = -1;
+  for (const step of [3, 2, 1, 0]) {
+    const indexBefore = before.indexOf(steps[step].text) - 3;
+    const indexAfter = after.indexOf(steps[step].text) - 3;
+    if (before[indexBefore] !== 'x' && after[indexAfter] === 'x') {
+      checked = step;
+      break;
+    }
+  }
```

Comment:
Would it make things easier if you change the checkboxes to something like `- [ ] <!-- STEP=investigating -->` and then only have to search by a known regex?

Reply:
Yes, but regex is hard... one moment.

Reply:
I love regex 🥺

Reply:
So my suggestion was to use an HTML comment and the step `status` field to make things easier - basically your regex would be something like: `new RegExp(\`^- [x] <!-- STEP=${step.status} -->\`)`, no need to match against the full text. wdyt?

Reply:
Makes sense and is a good idea 👍